src/bp/core/services/telemetry/hooks.ts

```diff
@@ -0,0 +1,86 @@
+import LicensingService from 'common/licensing-service'
+import { buildSchema } from 'common/telemetry'
+import Database from 'core/database'
+import { calculateHash } from 'core/misc/utils'
+import { TelemetryRepository } from 'core/repositories/telemetry'
+import { TYPES } from 'core/types'
+import { inject, injectable } from 'inversify'
+import _ from 'lodash'
+import ms from 'ms'
+
+import { GhostService } from '..'
+import { BotService } from '../bot-service'
+import { JobService } from '../job-service'
+
+import { TelemetryStats } from './telemetry-stats'
+
+interface BotHooks {
+  botId: string
+  [hookName: string]: any
+}
+
+interface HookPayload {
+  perBots: BotHooks[]
+  global: { [hookName: string]: number }
+}
+
+const BOT_HOOKS = [
+  'after_bot_mount',
+  'after_bot_unmount',
+  'after_event_processed',
+  'after_incoming_middleware',
+  'before_bot_import',
+  'before_incoming_middleware',
+  'before_outgoing_middleware',
+  'before_session_timeout',
+  'before_suggestions_election',
+  'on_bot_error'
+]
+
+const GLOBAL_HOOKS = ['after_server_start', 'after_stage_changed', 'on_incident_status_changed', 'on_stage_request']
+
+@injectable()
+export class HooksLifecycleStats extends TelemetryStats {
+  protected interval: number
+  protected url: string
+  protected lock: string
+
+  constructor(
+    @inject(TYPES.GhostService) ghostService: GhostService,
+    @inject(TYPES.Database) database: Database,
+    @inject(TYPES.LicensingService) licenseService: LicensingService,
+    @inject(TYPES.JobService) jobService: JobService,
+    @inject(TYPES.TelemetryRepository) telemetryRepo: TelemetryRepository,
+    @inject(TYPES.BotService) private botService: BotService
+  ) {
+    super(ghostService, database, licenseService, jobService, telemetryRepo)
+    this.url = process.TELEMETRY_URL
+    this.lock = 'botpress:telemetry-hooks'
+    this.interval = ms('1d')
+  }
+
+  protected async getStats() {
+    return {
+      ...buildSchema(await this.getServerStats(), 'server'),
+      event_type: 'hooks_lifecycle',
+      event_data: { schema: '1.0.0', lifeCycles: this.getHooksLifecycle() }
+    }
+  }
+
+  private async getHooksLifecycle(): Promise<HookPayload> {
+    const botIds = await this.botService.getBotsIds()
+    const perBots = await Promise.map(botIds, async id => {
+      const botHooksPaths = await this.ghostService.forBot(id).directoryListing('/hooks', '*.js')
```

Comment:
@spg @allardy @EFF @chsir17 Since we don't collect the name of the bots, should we at least collect whether the hooks are `builtin|custom` and `enabled|disabled` ?

Reply:
Good point. It think the payload should look more like this :
```
event_type: 'hooks'
event_data: {
  hooks: [
    {name: "the_hook_name.js", type: "built-in", "enabled": true, lifecycle: "before_outgoing_middleware"},
    {name: "redacted", type: "custom", "enabled": true, lifecycle: "after_event_processed"},
    ...
  ]
}
```

What do you guys think?

Reply:
The above ⬆️ payloadload gives us maximal resolution, which we can crunch down in the transformation pipeline in AWS.

Reply:
agreed !

Reply:
fixed: 77047f1ee25a284a0ee513a2c70e4b0aaa75b0bc